src/osd/osd_types.cc

```diff
@@ -4327,8 +4340,6 @@ void object_info_t::decode(bufferlist::iterator& bl)
     ::decode(omap_digest, bl);
   } else {
     data_digest = omap_digest = -1;
```

Comment:
I think we should call new_object here (and probably rename new_object to clear_digests).